grails-app/services/org/pih/warehouse/data/MigrationService.groovy

```diff
@@ -359,6 +399,128 @@ class MigrationService {
+
+
+        Map<String, List<String>> results = [:]
+        recordMigrationResult(results, transactions, true)
+
+        if (performMigration && transactions) {
+            List<Product> products = transactions.transactionEntries?.flatten()?.collect { TransactionEntry te -> te.inventoryItem.product }?.unique()
+            // Always create a stock snapshot before modifying any transactions in order to prevent quantity on hand
+            // differences for edge cases that were not addressed
+            log.debug("Creating inventory baseline for current stock for products found in old transactions")
+            Transaction currentInventoryBaseline = productInventoryTransactionMigrationService.createInventoryBaselineTransaction(
+                    location,
+                    null,
+                    products,
+                    null,
+                    "Inventory baseline created during old product inventory transactions migration",
+                    null,
+                    true,
+                    true
+            )
+
+            transactions.each { Transaction it ->
+                log.debug "Migrating transaction ${it.transactionNumber} with ${it.transactionEntries.size()} transaction entries"
+                List<TransactionEntry> entries = it.transactionEntries
+                // 2. Find all products in entries (and create inventory baseline for these)
+                List<Product> currentTransactionProducts = entries?.collect { TransactionEntry te -> te.inventoryItem.product }?.unique()
+
+                // 3. Create inventory baseline for old transaction that is being migrated (if enabled)
+                Map<String, AvailableItem> availableItems = productAvailabilityService.getAvailableItemsAtDateAsMap(
+                        location, currentTransactionProducts, it.transactionDate)
+
+                Transaction baselineTransaction = productInventoryTransactionMigrationService.createInventoryBaselineTransactionForGivenStock(
+                        location,
+                        null,
+                        availableItems.values(),
+                        it.transactionDate,
+                        "Migrated from single Product Inventory transaction to Baseline + Adjustment pair",
+                        null,
+                        // don't validate transaction date, there is old transaction at the same time, that will be removed
+                        false,
+                        true
+                )
+                if (baselineTransaction) {
+                    baselineTransaction.disableRefresh = true
+                    // Ugly workaround to omit auto-timestamping dateCreated
+                    // Needs to flush it first to be able to overwrite dateCreated
+                    baselineTransaction.save(flush: true)
+                    // Unfortunately, we need to use raw SQL to update the dateCreated field (otherwise it will be
+                    // auto-timestamped)
```

Comment:
I mean, it is true that date_created is changing to be the time of the migration. It represents the creation of the row in the database. Are there any scenarios where we can't simply rely on transaction_date? I know creation date will be used if two transactions have the same transaction date, but does that actually happen in practice? We really shouldn't allow that to happen imo since it would likely lead to confusion